src/v/cluster/rm_stm.h

```diff
@@ -300,6 +333,24 @@ class rm_stm final : public persisted_stm {
     bool _is_tx_enabled{false};
     ss::sharded<cluster::tx_gateway_frontend>& _tx_gateway_frontend;
     storage::snapshot_manager _abort_snapshot_mgr;
+
+    struct seq_entry_v0 {
+        model::producer_identity pid;
+        int32_t seq;
+        model::timestamp::type last_write_timestamp;
+    };
+
+    struct tx_snapshot_v0 {
+        static constexpr uint8_t version = 0;
+
+        std::vector<model::producer_identity> fenced;
+        std::vector<tx_range> ongoing;
+        std::vector<prepare_marker> prepared;
+        std::vector<tx_range> aborted;
+        std::vector<abort_index> abort_indexes;
+        model::offset offset;
+        std::vector<seq_entry_v0> seqs;
+    };
```

Comment:
it's probably reasonable to move these out of the header and into `rm_stm.cc` as they are now an internal detail.

Reply:
afaict these are only used in `rm_stm.cc` and could be added into a private namespace within that file and out of the the public header file.

Reply:
their existence is merely for serialization/deserialization now. so hiding them away would be clean